src/main.rs

```diff
@@ -236,6 +240,13 @@ mod app {
             telemetry.monitor.overtemp = gpio.overtemp();
             telemetry.monitor.poe = gpio.poe();
         });
+        // finalize temperature telemetry
+        for ch in InputChannel::into_enum_iter() {
+            telemetry.statistics[ch as usize] = c
+                .shared
+                .ch_statistics_buff
+                .lock(|buff| buff[ch as usize].into());
```

Comment:
Because `Buffer` is `Copy`, you are (AFAICT) not resetting here. `impl Default` and assign that (see below).

```rust
lock(|buff| {
    let stat = buff[ch as usize].into();
    buff[ch as usize] = Buffer::default();
    stat
}
```